### DISCUSSION TITLE
Вычитание исполнителей из плейлиста

### DISCUSSION THREAD

**Discussion by kisoid** [Chimildic/YaMuTools#56] [Идея] [answered]
В меню "Фильтр треков" можно добавить функцию "Вычитание исполнителей" по аналогии и с таким же интерфейсом, как в "Вычитание треков" (т.е. расширение соберёт список уникальных исполнителей по всем трекам плейлиста и потом надо будет проставить галки напротив тех исполнителей, которых надо удалить)

Только существующую функцию "Вычитание треков" в таком случае было бы логичнее переименовать в "Вычитание плейлистов".

---

_Кейс_: предположим, я создал плейлист, состоящий из большого количества новых для меня исполнителей (например, накидал в кучу разных сборников), потом послушал его некоторое время и осознал, что мне совершенно не нравятся треки _с участием_ Исполнителя_А.

Можно было бы, конечно, отсортировать плейлист по исполнителям, а затем вручную удалить лишние треки, но проблема в том, что Исполнитель_А может быть _не_ первичным исполнителем, т.е. вместо

`Исполнитель_А, Исполнитель_Б - Название_трека (Версия)`
может быть
`Исполнитель_Б, Исполнитель_А - Название_трека (Версия)`
или даже
`Исполнитель_Б, Исполнитель_В, Исполнитель_А - Название_трека (Версия)`
и т.д. ...

В таком случае с помощью сортировки и ручного удаления удастся только часть треков отсеять.

Опять же, иногда сортировка плейлиста в другом порядке нежелательна.

Ну и с новой функцией будет гораздо проще, если нежелательных исполнителей несколько или много треков придётся удалять...

**Comment by Chimildic** [comment, marked as answer]
Добавил нечто подобное в следующую версию расширения. Ожидайте обновления.

**Reply by kisoid** [reply]
> нечто подобное

Вот прямо так и хотелось! Спасибо, работает!

Единственное, что заметил, вот для этого трека: https://music.yandex.ru/album/964812/track/61111220
расширение не увидело второго исполнителя "Special EFX". Но я уверен, что тут проблема не с расширением, а с API, т.к. веб-морда Яндекса пишет "Chieli Minucci & Special EFX", в то время как API выдаёт только "Chieli Minucci" в качестве исполнителя...
